starlite/datastructures.py

```diff
@@ -72,13 +74,137 @@ def __init__(self, tasks: List[BackgroundTask]):
         super().__init__(tasks=tasks)
 
 
-class State(StarletteStateClass):
-    """An object that can be used to store arbitrary state.
+class State(Mapping[str, Any]):
```

Comment:
Is there a reason we aren't implementing as `MutableMapping` and define `__setitem__` and `__delitem__`?

Reply:
I started with this - I didnt like the KeyError / AttributeError ambiguity, but following your question I re-implemented this.